syft/tensor.py

```diff
@@ -1278,3 +1284,21 @@ def mv(tensormat, tensorvector):
             compatible with matrix {} '.format(tensorvector.data.shape, tensormat.data.shape))
     else:
         return TensorBase(np.matmul(tensormat.data, tensorvector.data))
+
+
+def masked_select(tensor, mask):
+    """
+    Returns a new 1D Tensor which indexes the ``input`` Tensor according to the binary mask ``mask``.
+    The shapes of the ``mask`` tensor and the ``input`` tensor don’t need to match, but they must be broadcastable.
+
+    :param tensor: Input tensor
+    :param mask: The binary mask (non-zero is treated as true)
+    :return: 1D output tensor
+    """
+    mask = _ensure_tensorbase(mask)
+    tensor = _ensure_tensorbase(tensor)
+    if tensor.encrypted or mask.encrypted:
+        raise NotImplemented
+    mask_broadcasted = np.broadcast_to(mask.data, tensor.data.shape)
```

Comment:
Broadcasting functionality is a bit different than the PyTorch version because this line makes broadcasting tensor => mask fail. Without this step I think it works as intended.

Reply:
Can you clarify what you mean?

Reply:
Tensor [1.0] & Mask [1.0,1.0,1.0] are broadcastable but fail in this PR.

Reply:
Oh! good catch. I will work on a fix.